src/test/gameworld.cpp

```diff
@@ -15,8 +27,102 @@ std::vector<std::string> FetchAndroidServerCommandQueue()
 	return FakeQueue;
 }
 
-TEST(GameWorld, Basic)
+class CGameWorldTestContext
 {
-	CGameWorld *pWorld = new CGameWorld();
-	delete pWorld;
+public:
+	IGameServer *m_pGameServer = nullptr;
+	IKernel *m_pKernel;
+	std::shared_ptr<CServerLogger> m_pServerLogger = nullptr;
+
+	CGameWorldTestContext()
+	{
+		std::vector<std::shared_ptr<ILogger>> vpLoggers;
+		std::shared_ptr<ILogger> pStdoutLogger;
+		if(pStdoutLogger)
+		{
+			vpLoggers.push_back(pStdoutLogger);
+		}
+		std::shared_ptr<CFutureLogger> pFutureFileLogger = std::make_shared<CFutureLogger>();
+		vpLoggers.push_back(pFutureFileLogger);
+		std::shared_ptr<CFutureLogger> pFutureConsoleLogger = std::make_shared<CFutureLogger>();
+		vpLoggers.push_back(pFutureConsoleLogger);
+		std::shared_ptr<CFutureLogger> pFutureAssertionLogger = std::make_shared<CFutureLogger>();
+		vpLoggers.push_back(pFutureAssertionLogger);
+
+		CServer *pServer = CreateServer();
+		pServer->SetLoggers(pFutureFileLogger, std::move(pStdoutLogger));
+
+		m_pKernel = IKernel::Create();
+		m_pKernel->RegisterInterface(pServer);
+
+		IEngine *pEngine = CreateEngine(GAME_NAME, pFutureConsoleLogger, (2 * std::thread::hardware_concurrency()) + 2);
+		m_pKernel->RegisterInterface(pEngine);
+
+		const char *apArgs[] = {"DDNet"};
+		IStorage *pStorage = CreateStorage(IStorage::EInitializationType::SERVER, 1, apArgs);
+		EXPECT_NE(pStorage, nullptr);
```

Comment:
```suggestion
		ASSERT_NE(pStorage, nullptr);
```
`EXPECT_*` is when you want the test to continue after the failed assertion, `ASSERT_` if you don't.

Reply:
Interesting, seems like that isn't possible: https://google.github.io/googletest/advanced.html#assertion-placement.

You cannot use `ASSERT_*` in functions not returning `void`. Then this test will run into a segmentation fault if `pStorage` is null. Oh well…